javascript/tables/src/validateTable.ts

```diff
@@ -0,0 +1,288 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type {
+	CodebookSchema,
+	ColumnError,
+	Field,
+	ValidationResult,
+} from '@datashaper/schema'
+import { DataType } from '@datashaper/schema'
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+
+export function validateTable(
+	table: ColumnTable,
+	codebook: CodebookSchema,
+): ValidationResult {
+	const validationResult: ValidationResult = {
+		columnErrors: [],
+	}
+
+	codebook.fields.forEach((field: Field) => {
+		const columnValues = table.array(field.name)
+		let flag = false
+		if (field.constraints != null) {
+			const constraints = field.constraints
+
+			//required constraint
+			if (!flag && constraints.required != null && constraints.required) {
```

Comment:
I'm not sure why these are checking `!= null` _and_ falsy check. You should be able to do just the latter.

Reply:
lint complains if you don't check first the nullability of the optional property and also the boolean